src/middlewared/middlewared/plugins/cluster_linux/ctdb_shared_volume.py

```diff
@@ -0,0 +1,232 @@
+                    'but is not a REPLICATE type volume. Please delete or rename '
+                    'this volume and try again.'
+                )
+            elif vol['replica'] < 3 or vol['num_bricks'] < 3:
+                raise CallError(
+                    f'A volume with the name "{self.CTDB_VOL_NAME}" already exists '
+                    'but is configured in a way that could cause split-brain. '
+                    'Please delete or rename this volume and try again.'
+                )
+            elif vol['status'] != 'Started':
+                exists = True
+                start = await self.middleware.call('gluster.volume.start', self.CTDB_VOL_NAME)
+                await start.wait()
+                if start.error:
+                    raise CallError(f'{start.error}')
+                started = True
+            else:
+                exists = started = True
+
+        return exists, started
+
+    @accepts()
+    @job(lock='ctdb_create_lock')
+    async def create(self, job):
+
+        """
+        Create and mount the shared volume to be used
+        by ctdb daemon.
+        """
+
+        # check if ctdb shared volume already exists and started
+        exists, started = await self.shared_volume_exists_and_started()
+        if exists and started:
+            return
+
+        if not exists:
+            # get the peers in the TSP
+            peers = await self.middleware.call('gluster.peer.pool')
+            await peers.wait()
+            if peers.error:
+                raise CallError(f'{peers.error}')
+            elif not peers.result:
+                raise CallError('No peers detected')
+
+            # shared storage volume requires 3 nodes, minimally, to
+            # prevent the dreaded split-brain
+            con_peers = [i['hostname'] for i in peers.result if i['connected'] == 'Connected']
+            if len(con_peers) < 3:
+                raise CallError(
+                    '3 peers must be present and connected before the ctdb '
+                    'shared volume can be created.'
+                )
+
+            # create the ctdb shared volume
+            req = await self.construct_gluster_ctdb_api_request(con_peers)
+            create = await self.middleware.call('gluster.volume.create', req)
+            await create.wait()
+            if create.error:
+                raise CallError(f'{create.error}')
+
+        if not started:
+            # start it if we get here
+            start = await self.middleware.call('gluster.volume.start', self.CTDB_VOL_NAME)
+            await start.wait()
+            if start.error:
+                raise CallError(f'{start.error}')
+
+        # try to mount it locally
+        mount = await self.middleware.call('ctdb.shared.volume.mount')
+        await mount.wait()
+        if mount.error:
+            raise CallError(f'{mount.error}')
+
+        return 'SUCCESS'
+
+    @accepts()
+    @job(lock='ctdb_delete_lock')
```

Comment:
Shouldn't it be the same lock everywhere?

Reply:
The create/delete methods need to share the same lock but we need to use a separate lock for mount/umount methods since those are called from the create/delete methods respectively.